Fixed #31552 -- Added support for LZMA and XZ fixtures to loaddata.

pauloxnet opened the pull request:
https://code.djangoproject.com/ticket/31552

felixxm commented on docs/ref/django-admin.txt:
```diff
@@ -601,12 +601,13 @@ The :djadmin:`dumpdata` command can be used to generate input for ``loaddata``.
-would look for any of ``mydata.json``, ``mydata.json.zip``,
-``mydata.json.gz``, or ``mydata.json.bz2``. The first file contained within a
+would look for any of ``mydata.json``, ``mydata.json.zip``, ``mydata.json.gz``,
+``mydata.json.bz2`` or ``mydata.json.xz``. The first file contained within a
```
Serial comma should stay before 'or' (see https://en.wikipedia.org/wiki/Serial_comma).

felixxm replied:
`versionchanged`  annotation is missing, e.g.:
```
.. versionchanged:: 3.2

    Support for XZ archives (``.xz``) and LZMA archives (``.lzma``) was added.
```

felixxm commented on django/core/management/commands/loaddata.py:
```diff
@@ -97,6 +103,8 @@ def loaddata(self, fixture_labels):
         if has_bz2:
             self.compression_formats['bz2'] = (bz2.BZ2File, 'r')
+        if has_lzma:
+            self.compression_formats['xz'] = (lzma.LZMAFile, 'r')
```
I think we should add also `lzma` format.

felixxm commented on docs/ref/django-admin.txt:
```diff
@@ -601,12 +601,13 @@ The :djadmin:`dumpdata` command can be used to generate input for ``loaddata``.
 ~~~~~~~~~~~~~~~~~~~
 
-Fixtures may be compressed in ``zip``, ``gz``, or ``bz2`` format. For example::
+Fixtures may be compressed in ``zip``, ``gz``, ``bz2`` or ``xz`` format.
```
Serial comma should stay before 'or' (see https://en.wikipedia.org/wiki/Serial_comma).

felixxm commented on tests/fixtures/tests.py:
```diff
@@ -559,6 +565,13 @@ def test_compressed_loading_bz2(self):
+        management.call_command('loaddata', 'fixture5.json.xz', verbosity=0)
+        self.assertQuerysetEqual(Article.objects.all(), [
+            '<Article: WoW subscribers now outnumber readers>',
+        ])
```
Please add test also for `.lzma` archive.

pauloxnet commented:
@felixxm thanks for your review int he updated commit I added support for LZMA archives , fixes documentation issues and added a section in 3.2 release notes.

felixxm commented:
@pauloxnet Thanks :+1:

pauloxnet commented:
@felixxm Thanks for your review and merge. 👌
